clang/lib/Sema/SemaType.cpp

```diff
@@ -2727,8 +2731,15 @@ QualType Sema::BuildVectorType(QualType CurType, Expr *SizeExpr,
   }
 
   if (!TypeSize || VectorSizeBits % TypeSize) {
-    Diag(AttrLoc, diag::err_attribute_invalid_size)
-        << SizeExpr->getSourceRange();
+    // Disable errors for SYCL and OpenMP device since definition of __bf16 is
+    // being moved to a shared header and it causes new errors emitted when
+    // host code is compiled with device compiler for SPIR target.
+    // FIXME: device code specific diagnostic is probably needed.
+    if (!(!TypeSize &&
+          (getLangOpts().OpenMPIsDevice || getLangOpts().SYCLIsDevice))) {
```

Comment:
Can you double check if `OpenMPIsDevice` switch is required here? It may not.

Reply:
@preethi-intel, have you open a ticket for this FIXME to follow up?

Reply:
@pvchupin , I have opened an internal tracker for this FIXME.